backend/apps/beaver_api/beaver_api/code_api/views.py

```diff
@@ -19,3 +24,52 @@ class GetRandomCodeDocumentView(GenericAPIView, GetRandomObjectMixin):
 
     def get(self, request: Request, *args, **kwargs) -> Response:
         return self.get_random_document(request, *args, **kwargs)
+
+class CodeDocumentCreateView(generics.CreateAPIView):
+    serializer_class = CodeDocumentSerializer
+
+    def post(self, request: Request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data, many=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
```

Comment:
The problem with that approach is for each item in the list of `request.data` you will call DB. It is better to use `bulk_create` to save all instances at once.  Check the [doc](https://www.django-rest-framework.org/api-guide/serializers/#customizing-listserializer-behavior)